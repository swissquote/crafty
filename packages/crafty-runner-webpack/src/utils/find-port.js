const net = require("net");
const path = require("path");

const hash = require("hash-index");

function servicePort(name) {
  let i = 0;
  const reservedPorts = require("./reservedPorts.json");
  const range = [];

  for (let n = 1024; n <= 65535; n++) {
    if (reservedPorts[i] === n) {
      i++;
    } else {
      range.push(n);
    }
  }

  const index = hash(name, range.length);

  return range[index];
}

const portRange = 100;
const assigned = {};

function isPortTaken(port, fn) {
  const tester = net
    .createServer()
    .once("error", (err) => {
      if (err.code === "EADDRINUSE") {
        fn(null, true);
      } else {
        fn(err);
      }
    })
    .once("listening", () => {
      tester
        .once("close", () => {
          fn(null, false);
        })
        .close();
    })
    .listen(port);
}

function parse(arr, status, cb) {
  if (arr.length === 0) {
    cb(null, false);
  }

  const port = arr.shift();

  isPortTaken(port, (err, taken) => {
    //this should send error to user but it'll stop searching for ports...
    if (taken === status) {
      cb(null, port); //It's open
    } else {
      setImmediate(() => parse(arr, status, cb));
    }
  });
}

function findOpen(start, cb) {
  const stop = start + portRange;

  const ports = [];

  for (let i = start; i < stop; i++) {
    ports.push(i);
  }

  return parse(ports, false, cb);
}

module.exports = {
  getFree(buildname) {
    if (assigned[buildname]) {
      return Promise.resolve(assigned[buildname]);
    }

    const basePort = servicePort(path.join(__dirname, buildname));

    return new Promise((resolve, reject) => {
      findOpen(basePort, (err, port) => {
        if (err) {
          reject(err);
          return;
        }

        assigned[buildname] = port;

        resolve(port);
      });
    });
  },

  release(buildname) {
    delete assigned[buildname];
  },
};
