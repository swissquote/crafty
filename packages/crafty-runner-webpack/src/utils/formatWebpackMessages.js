/**
 * Copyright (c) 2015-present, Facebook, Inc.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree. An additional grant
 * of patent rights can be found in the PATENTS file in the same directory.
 */

// WARNING: this code is untranspiled and is used in browser too.
// Please make sure any changes are in ES5 or contribute a Babel compile step.

// Some custom utilities to prettify Webpack output.
// This is quite hacky and hopefully won't be needed when Webpack fixes this.
// https://github.com/webpack/webpack/issues/2878

const colors = require("ansi-colors");

const friendlySyntaxErrorLabel = "Syntax error:";

function isLikelyASyntaxError(message) {
  return message.indexOf(friendlySyntaxErrorLabel) !== -1;
}

// Cleans up webpack error messages.
// eslint-disable-next-line no-unused-vars
function formatMessage(originalMessage, isError) {
  let lines = (originalMessage.message || originalMessage).split("\n");

  // Strip Webpack-added headers off errors/warnings
  // https://github.com/webpack/webpack/blob/master/lib/ModuleError.js
  lines = lines.filter(line => !/Module [A-z ]+\(from/.test(line));

  // Transform parsing error into syntax error
  lines = lines.map(line => {
    const parsingError = /Line (\d+):(?:(\d+):)?\s*Parsing error: (.+)$/.exec(
      line
    );
    if (!parsingError) {
      return line;
    }
    const [, errorLine, errorColumn, errorMessage] = parsingError;
    return `${friendlySyntaxErrorLabel} ${errorMessage} (${errorLine}:${errorColumn})`;
  });

  let message = lines.join("\n");
  // Smoosh syntax errors (commonly found in CSS)
  message = message.replace(
    /SyntaxError\s+\((\d+):(\d+)\)\s*(.+?)\n/g,
    `${friendlySyntaxErrorLabel} $3 ($1:$2)\n`
  );
  // Remove columns from ESLint formatter output (we added these for more
  // accurate syntax errors)
  message = message.replace(/Line (\d+):\d+:/g, "Line $1:");
  // Clean up export errors
  message = message.replace(
    /^.*export '(.+?)' was not found in '(.+?)'.*$/gm,
    "Attempted import error: '$1' is not exported from '$2'."
  );
  message = message.replace(
    /^.*export 'default' \(imported as '(.+?)'\) was not found in '(.+?)'.*$/gm,
    "Attempted import error: '$2' does not contain a default export (imported as '$1')."
  );
  message = message.replace(
    /^.*export '(.+?)' \(imported as '(.+?)'\) was not found in '(.+?)'.*$/gm,
    "Attempted import error: '$1' is not exported from '$3' (imported as '$2')."
  );
  lines = message.split("\n");

  // Remove leading newline
  if (lines.length > 2 && lines[1].trim() === "") {
    lines.splice(1, 1);
  }
  // Clean up file name
  lines[0] = lines[0].replace(/^(.*) \d+:\d+-\d+$/, "$1");

  lines = lines.filter(line => {
    // Webpack adds a list of entry points to warning messages:
    //  @ ./src/index.js
    //  @ multi react-scripts/~/react-dev-utils/webpackHotDevClient.js ...
    // It is misleading (and unrelated to the warnings) so we clean it up.
    // It is only useful for syntax errors but we have beautiful frames for them.
    return line.indexOf(" @ ") !== 0;
  });

  // Cleans up verbose "module not found" messages for files and packages.
  if (lines[1] && lines[1].indexOf("Module not found: ") === 0) {
    lines = [
      lines[0],
      lines[1]
        .replace("Error: ", "")
        .replace("Module not found: Cannot find file:", "Cannot find file:")
    ];
  }

  lines[0] = colors.inverse(lines[0]);

  message = lines.join("\n");
  // Internal stacks are generally useless so we strip them... with the
  // exception of stacks containing `webpack:` because they're normally
  // from user code generated by Webpack. For more information see
  // https://github.com/facebook/create-react-app/pull/1050
  message = message.replace(
    /^\s*at\s((?!webpack:).)*:\d+:\d+[\s)]*(\n|$)/gm,
    ""
  ); // at ... ...:x:y
  message = message.replace(/^\s*at\s<anonymous>(\n|$)/gm, ""); // at <anonymous>
  lines = message.split("\n");

  // Remove duplicated newlines
  lines = lines.filter(
    (line, index, arr) =>
      index === 0 || line.trim() !== "" || line.trim() !== arr[index - 1].trim()
  );

  // Reassemble the message
  message = lines.join("\n");
  return message.trim();
}

function formatWebpackMessages(json) {
  const formattedErrors = json.errors.map(message =>
    formatMessage(message, true)
  );
  const formattedWarnings = json.warnings.map(message =>
    formatMessage(message, false)
  );
  const result = {
    errors: formattedErrors,
    warnings: formattedWarnings
  };
  if (result.errors.some(isLikelyASyntaxError)) {
    // If there are any syntax errors, show just them.
    // This prevents a confusing ESLint parsing error
    // preceding a much more useful Babel syntax error.
    result.errors = result.errors.filter(isLikelyASyntaxError);
  }
  return result;
}

module.exports = formatWebpackMessages;
