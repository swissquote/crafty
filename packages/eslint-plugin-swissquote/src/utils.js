// Eslint can't load plugins transitively (from a shared config)
// So we have to include the file ourselves and include the rules as if they were ours.
// Solution proposed by @nzakas himself : https://github.com/eslint/eslint/issues/3458#issuecomment-257161846
// replaces `extends: "plugin:react/recommended",`

function addMissingRules(source, destination, except = []) {
  Object.keys(source).forEach(ruleName => {
    // Only define the rules we don't have configured yet
    const key =
      ruleName.indexOf("/") > -1
        ? `@swissquote/swissquote/${ruleName}`
        : ruleName;
    if (!destination.hasOwnProperty(key) && except.indexOf(key) === -1) {
      destination[key] = source[ruleName];
    }
  });
}

function warn() {
  // When running in development mode, some errors can just be warnings.
  // Some errors don't need to break the build if they aren't threatening the functionality.
  return process.env.NODE_ENV === "development" ? "warn" : "error";
}

module.exports = {
  addMissingRules,
  warn
};
