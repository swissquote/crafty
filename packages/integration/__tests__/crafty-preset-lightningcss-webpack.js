import { test, expect } from "vitest";
import * as testUtils from "../utils.js";

const BUNDLE = "dist/js/myBundle.min.js";
const BUNDLE_MAP = `${BUNDLE}.map`;

test("Compiles CSS within webpack", async () => {
  const cwd = await testUtils.getCleanFixtures(
    "crafty-preset-lightningcss-webpack/compiles"
  );

  const result = await testUtils.run(["run", "default"], cwd);

  expect(result).toMatchSnapshot();
  expect(result.status).toBe(0);

  expect(testUtils.exists(cwd, BUNDLE)).toBeTruthy();
  expect(testUtils.exists(cwd, BUNDLE_MAP)).toBeTruthy();

  expect(testUtils.readForSnapshot(cwd, BUNDLE)).toMatchSnapshot();
});

test("Fails gracefully on broken markup", async () => {
  const cwd = await testUtils.getCleanFixtures(
    "crafty-preset-lightningcss-webpack/fails"
  );

  const result = await testUtils.run(["run", "default"], cwd);

  expect(result).toMatchSnapshot();
  expect(result.status).toBe(1);

  expect(testUtils.exists(cwd, BUNDLE)).toBeFalsy();
  expect(testUtils.exists(cwd, BUNDLE_MAP)).toBeFalsy();
});

test(
  "Compiles CSS within webpack, extracts CSS ('extractCSS' boolean option)",
  async () => {
    const cwd = await testUtils.getCleanFixtures(
      "crafty-preset-lightningcss-webpack/extract-boolean"
    );

    const result = await testUtils.run(["run", "default"], cwd);

    expect(result).toMatchSnapshot();
    expect(result.status).toBe(0);

    expect(testUtils.exists(cwd, BUNDLE)).toBeTruthy();
    expect(testUtils.exists(cwd, BUNDLE_MAP)).toBeTruthy();
    expect(testUtils.exists(cwd, "dist/js/myBundle-default.min.css")).toBeTruthy();
    expect(testUtils.exists(cwd, "dist/js/myBundle-default.min.css")).toBeTruthy();
    expect(testUtils.exists(cwd, "dist/js/myBundle-default.min.css.map")).toBeTruthy();
    expect(testUtils.readForSnapshot(cwd, BUNDLE)).toMatchSnapshot();
    expect(testUtils.readForSnapshot(cwd, BUNDLE)).toMatchSnapshot();
    expect(
      testUtils.readForSnapshot(cwd, "dist/js/myBundle-default.min.css")
    ).toMatchSnapshot();
  }
);

test(
  "Compiles CSS within webpack, extracts CSS ('extractCSS' string option)",
  async () => {
    const cwd = await testUtils.getCleanFixtures(
      "crafty-preset-lightningcss-webpack/extract-string"
    );

    const result = await testUtils.run(["run", "default"], cwd);

    expect(result).toMatchSnapshot();
    expect(result.status).toBe(0);

    expect(testUtils.exists(cwd, BUNDLE)).toBeTruthy();
    expect(testUtils.exists(cwd, BUNDLE_MAP)).toBeTruthy();
    expect(testUtils.exists(cwd, "dist/js/myBundle-string.min.css")).toBeTruthy();
    expect(testUtils.exists(cwd, "dist/js/myBundle-string.min.css.map")).toBeTruthy();

    expect(testUtils.readForSnapshot(cwd, BUNDLE)).toMatchSnapshot();
    expect(testUtils.readForSnapshot(cwd, BUNDLE)).toMatchSnapshot();
    expect(
      testUtils.readForSnapshot(cwd, "dist/js/myBundle-string.min.css")
    ).toMatchSnapshot();
  }
);

test(
  "Compiles CSS within webpack, extracts CSS ('extractCSS' object option)",
  async () => {
    const cwd = await testUtils.getCleanFixtures(
      "crafty-preset-lightningcss-webpack/extract-object"
    );

    const result = await testUtils.run(["run", "default"], cwd);

    expect(result).toMatchSnapshot();
    expect(result.status).toBe(0);

    expect(testUtils.exists(cwd, BUNDLE)).toBeTruthy();
    expect(testUtils.exists(cwd, BUNDLE_MAP)).toBeTruthy();
    expect(testUtils.exists(cwd, "dist/js/myBundle-object.min.css")).toBeTruthy();
    expect(testUtils.exists(cwd, "dist/js/myBundle-object.min.css.map")).toBeTruthy();

    expect(testUtils.readForSnapshot(cwd, BUNDLE)).toMatchSnapshot();
    expect(testUtils.readForSnapshot(cwd, BUNDLE)).toMatchSnapshot();
    expect(
      testUtils.readForSnapshot(cwd, "dist/js/myBundle-object.min.css")
    ).toMatchSnapshot();
  }
);
