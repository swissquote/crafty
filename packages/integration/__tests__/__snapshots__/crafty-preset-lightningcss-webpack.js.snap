// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Compiles CSS within webpack 1`] = `
{
  "status": 0,
  "stdall": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            css: __PATH__/packages/integration-fixtures-cjs/fixtures/crafty-preset-lightningcss-webpack/compiles/dist/css
           js: __PATH__/packages/integration-fixtures-cjs/fixtures/crafty-preset-lightningcss-webpack/compiles/dist/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myBundle' ...
asset myBundle.min.js ___ KiB [emitted] [minimized] (name: default)
  sourceMap myBundle.min.js.map ___ KiB [emitted] [dev] (auxiliary name: default)
Entrypoint default ___ KiB (___ KiB) = myBundle.min.js 1 auxiliary asset
chunk (runtime: default) myBundle.min.js (default) ___ KiB (javascript) ___ KiB (runtime) [entry] [rendered]
  dependent modules ___ KiB [dependent] 10 modules
  runtime modules ___ KiB 5 modules
  ./js/app.js + 1 modules ___ KiB [built] [code generated]

  Compiled successfully!
  Δt ____ms

[__:__:__] Finished 'js_myBundle' after ____ ms
[__:__:__] Finished 'js' after ____ ms
[__:__:__] Finished 'default' after ____ ms
",
}
`;

exports[`Compiles CSS within webpack 2`] = `
"!function(e,n){if("object"==typeof exports&&"object"==typeof module)module.exports=n();else if("function"==typeof define&&define.amd)define([],n);else{var t=n();for(var r in t)("object"==typeof exports?exports:e)[r]=t[r]}}(self,function(){return function(){"use strict";var e={113:function(e){e.exports=function(e){if("undefined"==typeof document)return{update:function(){},remove:function(){}};var n=e.insertStyleElement(e);return{update:function(t){var r,o,a;r="",t.supports&&(r+="@supports (".concat(t.supports,") {")),t.media&&(r+="@media ".concat(t.media," {")),(o=void 0!==t.layer)&&(r+="@layer".concat(t.layer.length>0?" ".concat(t.layer):""," {")),r+=t.css,o&&(r+="}"),t.media&&(r+="}"),t.supports&&(r+="}"),(a=t.sourceMap)&&"undefined"!=typeof btoa&&(r+="\\n/*# sourceMappingURL=data:application/json;base64,".concat(btoa(unescape(encodeURIComponent(JSON.stringify(a))))," */")),e.styleTagTransform(r,n,e.options)},remove:function(){!function(e){if(null!==e.parentNode)e.parentNode.removeChild(e)}(n)}}}},184:function(e){var n=[];function t(e){for(var t=-1,r=0;r<n.length;r++)if(n[r].identifier===e){t=r;break}return t}function r(e,r){for(var o={},a=[],i=0;i<e.length;i++){var c=e[i],u=r.base?c[0]+r.base:c[0],s=o[u]||0,f="".concat(u," ").concat(s);o[u]=s+1;var l=t(f),d={css:c[1],media:c[2],sourceMap:c[3],supports:c[4],layer:c[5]};if(-1!==l)n[l].references++,n[l].updater(d);else{var p=function(e,n){var t=n.domAPI(n);return t.update(e),function(n){n?(n.css!==e.css||n.media!==e.media||n.sourceMap!==e.sourceMap||n.supports!==e.supports||n.layer!==e.layer)&&t.update(e=n):t.remove()}}(d,r);r.byIndex=i,n.splice(i,0,{identifier:f,updater:p,references:1})}a.push(f)}return a}e.exports=function(e,o){var a=r(e=e||[],o=o||{});return function(e){e=e||[];for(var i=0;i<a.length;i++){var c=t(a[i]);n[c].references--}for(var u=r(e,o),s=0;s<a.length;s++){var f=t(a[s]);0===n[f].references&&(n[f].updater(),n.splice(f,1))}a=u}}},219:function(e){var n={};e.exports=function(e,t){var r=function(e){if(void 0===n[e]){var t=document.querySelector(e);if(window.HTMLIFrameElement&&t instanceof window.HTMLIFrameElement)try{t=t.contentDocument.head}catch(e){t=null}n[e]=t}return n[e]}(e);if(!r)throw Error("Couldn't find a style target. This probably means that the value for the 'insert' parameter is invalid.");r.appendChild(t)}},332:function(e){e.exports=function(e){var n=document.createElement("style");return e.setAttributes(n,e.attributes),e.insert(n,e.options),n}},401:function(e){e.exports=function(e,n){if(n.styleSheet)n.styleSheet.cssText=e;else{for(;n.firstChild;)n.removeChild(n.firstChild);n.appendChild(document.createTextNode(e))}}},416:function(e,n,t){e.exports=function(e){var n=t.nc;n&&e.setAttribute("nonce",n)}},599:function(e,n,t){var r=t(986),o=t.n(r),a=t(659),i=t.n(a)()(o());i.push([e.id,"a {\\n  color: #00f;\\n}\\n",""]),n.A=i},659:function(e){e.exports=function(e){var n=[];return n.toString=function(){return this.map(function(n){var t="",r=void 0!==n[5];return n[4]&&(t+="@supports (".concat(n[4],") {")),n[2]&&(t+="@media ".concat(n[2]," {")),r&&(t+="@layer".concat(n[5].length>0?" ".concat(n[5]):""," {")),t+=e(n),r&&(t+="}"),n[2]&&(t+="}"),n[4]&&(t+="}"),t}).join("")},n.i=function(e,t,r,o,a){"string"==typeof e&&(e=[[null,e,void 0]]);var i={};if(r)for(var c=0;c<this.length;c++){var u=this[c][0];null!=u&&(i[u]=!0)}for(var s=0;s<e.length;s++){var f=[].concat(e[s]);r&&i[f[0]]||(void 0!==a&&(void 0===f[5]||(f[1]="@layer".concat(f[5].length>0?" ".concat(f[5]):""," {").concat(f[1],"}")),f[5]=a),t&&(f[2]&&(f[1]="@media ".concat(f[2]," {").concat(f[1],"}")),f[2]=t),o&&(f[4]?(f[1]="@supports (".concat(f[4],") {").concat(f[1],"}"),f[4]=o):f[4]="".concat(o)),n.push(f))}},n}},916:function(e,n,t){var r=t(986),o=t.n(r),a=t(659),i=t.n(a),c=t(599),u=i()(o());u.i(c.A),u.push([e.id,".Parent {\\n  background: #fff;\\n}\\n\\n.Parent--before {\\n  color: #333;\\n}\\n\\n.Parent .Child {\\n  background: #000;\\n}\\n\\n.Parent--after {\\n  color: #eee;\\n}\\n\\n:root {\\n  --color: red;\\n}\\n\\n.MenuLink {\\n  color: var(--color);\\n}\\n",""]),n.A=u},986:function(e){e.exports=function(e){return e[1]}}},n={};function t(r){var o=n[r];if(void 0!==o)return o.exports;var a=n[r]={id:r,exports:{}};return e[r](a,a.exports,t),a.exports}t.n=function(e){var n=e&&e.__esModule?function(){return e.default}:function(){return e};return t.d(n,{a:n}),n},t.d=function(e,n){for(var r in n)t.o(n,r)&&!t.o(e,r)&&Object.defineProperty(e,r,{enumerable:!0,get:n[r]})},t.o=function(e,n){return Object.prototype.hasOwnProperty.call(e,n)},t.r=function(e){"undefined"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(e,"__esModule",{value:!0})},t.nc=void 0;var r={};t.r(r),t.d(r,{default:function(){return b}});var o=t(184),a=t.n(o),i=t(113),c=t.n(i),u=t(219),s=t.n(u),f=t(416),l=t.n(f),d=t(332),p=t.n(d),v=t(401),m=t.n(v),y=t(916),h={};h.styleTagTransform=m(),h.setAttributes=l(),h.insert=s().bind(null,"head"),h.domAPI=c(),h.insertStyleElement=p(),a()(y.A,h),y.A&&y.A.locals&&y.A.locals;class b{}return r}()});
//# sourceMappingURL=myBundle.min.js.map"
`;

exports[`Compiles CSS within webpack, extracts CSS ('extractCSS' boolean option) 1`] = `
{
  "status": 0,
  "stdall": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            css: __PATH__/packages/integration-fixtures-cjs/fixtures/crafty-preset-lightningcss-webpack/extract-boolean/dist/css
           js: __PATH__/packages/integration-fixtures-cjs/fixtures/crafty-preset-lightningcss-webpack/extract-boolean/dist/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myBundle' ...
asset myBundle.min.js ___ KiB [emitted] [minimized] (name: default)
  sourceMap myBundle.min.js.map ___ KiB [emitted] [dev] (auxiliary name: default)
asset myBundle-default.min.css ___ KiB [emitted] [minimized] (name: default)
  sourceMap myBundle-default.min.css.map ___ KiB [emitted] [dev] (auxiliary name: default)
Entrypoint default ___ KiB (___ KiB) = myBundle-default.min.css ___ KiB myBundle.min.js ___ KiB 2 auxiliary assets
chunk (runtime: default) myBundle-default.min.css, myBundle.min.js (default) ___ KiB (javascript) ___ KiB (css/mini-extract) ___ KiB (runtime) [entry] [rendered]
  runtime modules ___ KiB 3 modules
  dependent modules ___ KiB [dependent] 2 modules
  ./js/app.js + 1 modules ___ KiB [built] [code generated]

  Compiled successfully!
  Δt ____ms

[__:__:__] Finished 'js_myBundle' after ____ ms
[__:__:__] Finished 'js' after ____ ms
[__:__:__] Finished 'default' after ____ ms
",
}
`;

exports[`Compiles CSS within webpack, extracts CSS ('extractCSS' boolean option) 2`] = `
"!function(e,t){if("object"==typeof exports&&"object"==typeof module)module.exports=t();else if("function"==typeof define&&define.amd)define([],t);else{var o=t();for(var n in o)("object"==typeof exports?exports:e)[n]=o[n]}}(self,function(){return function(){"use strict";var e={};e.d=function(t,o){for(var n in o)e.o(o,n)&&!e.o(t,n)&&Object.defineProperty(t,n,{enumerable:!0,get:o[n]})},e.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},e.r=function(e){"undefined"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(e,"__esModule",{value:!0})};var t={};e.r(t),e.d(t,{default:function(){return o}});class o{}return t}()});
//# sourceMappingURL=myBundle.min.js.map"
`;

exports[`Compiles CSS within webpack, extracts CSS ('extractCSS' boolean option) 3`] = `
"!function(e,t){if("object"==typeof exports&&"object"==typeof module)module.exports=t();else if("function"==typeof define&&define.amd)define([],t);else{var o=t();for(var n in o)("object"==typeof exports?exports:e)[n]=o[n]}}(self,function(){return function(){"use strict";var e={};e.d=function(t,o){for(var n in o)e.o(o,n)&&!e.o(t,n)&&Object.defineProperty(t,n,{enumerable:!0,get:o[n]})},e.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},e.r=function(e){"undefined"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(e,"__esModule",{value:!0})};var t={};e.r(t),e.d(t,{default:function(){return o}});class o{}return t}()});
//# sourceMappingURL=myBundle.min.js.map"
`;

exports[`Compiles CSS within webpack, extracts CSS ('extractCSS' boolean option) 4`] = `
"a{color:#00f}.Parent{background:#fff}.Parent--before{color:#333}.Parent .Child{background:#000}.Parent--after{color:#eee}:root{--color:red}.MenuLink{color:var(--color)}
/*# sourceMappingURL=myBundle-default.min.css.map*/"
`;

exports[`Compiles CSS within webpack, extracts CSS ('extractCSS' object option) 1`] = `
{
  "status": 0,
  "stdall": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            css: __PATH__/packages/integration-fixtures-cjs/fixtures/crafty-preset-lightningcss-webpack/extract-object/dist/css
           js: __PATH__/packages/integration-fixtures-cjs/fixtures/crafty-preset-lightningcss-webpack/extract-object/dist/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myBundle' ...
asset myBundle.min.js ___ KiB [emitted] [minimized] (name: default)
  sourceMap myBundle.min.js.map ___ KiB [emitted] [dev] (auxiliary name: default)
asset myBundle-object.min.css ___ KiB [emitted] [minimized] (name: default)
  sourceMap myBundle-object.min.css.map ___ KiB [emitted] [dev] (auxiliary name: default)
Entrypoint default ___ KiB (___ KiB) = myBundle-object.min.css ___ KiB myBundle.min.js ___ KiB 2 auxiliary assets
chunk (runtime: default) myBundle-object.min.css, myBundle.min.js (default) ___ KiB (javascript) ___ KiB (css/mini-extract) ___ KiB (runtime) [entry] [rendered]
  runtime modules ___ KiB 3 modules
  dependent modules ___ KiB [dependent] 2 modules
  ./js/app.js + 1 modules ___ KiB [built] [code generated]

  Compiled successfully!
  Δt ____ms

[__:__:__] Finished 'js_myBundle' after ____ ms
[__:__:__] Finished 'js' after ____ ms
[__:__:__] Finished 'default' after ____ ms
",
}
`;

exports[`Compiles CSS within webpack, extracts CSS ('extractCSS' object option) 2`] = `
"!function(e,t){if("object"==typeof exports&&"object"==typeof module)module.exports=t();else if("function"==typeof define&&define.amd)define([],t);else{var o=t();for(var n in o)("object"==typeof exports?exports:e)[n]=o[n]}}(self,function(){return function(){"use strict";var e={};e.d=function(t,o){for(var n in o)e.o(o,n)&&!e.o(t,n)&&Object.defineProperty(t,n,{enumerable:!0,get:o[n]})},e.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},e.r=function(e){"undefined"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(e,"__esModule",{value:!0})};var t={};e.r(t),e.d(t,{default:function(){return o}});class o{}return t}()});
//# sourceMappingURL=myBundle.min.js.map"
`;

exports[`Compiles CSS within webpack, extracts CSS ('extractCSS' object option) 3`] = `
"!function(e,t){if("object"==typeof exports&&"object"==typeof module)module.exports=t();else if("function"==typeof define&&define.amd)define([],t);else{var o=t();for(var n in o)("object"==typeof exports?exports:e)[n]=o[n]}}(self,function(){return function(){"use strict";var e={};e.d=function(t,o){for(var n in o)e.o(o,n)&&!e.o(t,n)&&Object.defineProperty(t,n,{enumerable:!0,get:o[n]})},e.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},e.r=function(e){"undefined"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(e,"__esModule",{value:!0})};var t={};e.r(t),e.d(t,{default:function(){return o}});class o{}return t}()});
//# sourceMappingURL=myBundle.min.js.map"
`;

exports[`Compiles CSS within webpack, extracts CSS ('extractCSS' object option) 4`] = `
"a{color:#00f}.Parent{background:#fff}.Parent--before{color:#333}.Parent .Child{background:#000}.Parent--after{color:#eee}:root{--color:red}.MenuLink{color:var(--color)}
/*# sourceMappingURL=myBundle-object.min.css.map*/"
`;

exports[`Compiles CSS within webpack, extracts CSS ('extractCSS' string option) 1`] = `
{
  "status": 0,
  "stdall": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            css: __PATH__/packages/integration-fixtures-cjs/fixtures/crafty-preset-lightningcss-webpack/extract-string/dist/css
           js: __PATH__/packages/integration-fixtures-cjs/fixtures/crafty-preset-lightningcss-webpack/extract-string/dist/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myBundle' ...
asset myBundle.min.js ___ KiB [emitted] [minimized] (name: default)
  sourceMap myBundle.min.js.map ___ KiB [emitted] [dev] (auxiliary name: default)
asset myBundle-string.min.css ___ KiB [emitted] [minimized] (name: default)
  sourceMap myBundle-string.min.css.map ___ KiB [emitted] [dev] (auxiliary name: default)
Entrypoint default ___ KiB (___ KiB) = myBundle-string.min.css ___ KiB myBundle.min.js ___ KiB 2 auxiliary assets
chunk (runtime: default) myBundle-string.min.css, myBundle.min.js (default) ___ KiB (javascript) ___ KiB (css/mini-extract) ___ KiB (runtime) [entry] [rendered]
  runtime modules ___ KiB 3 modules
  dependent modules ___ KiB [dependent] 2 modules
  ./js/app.js + 1 modules ___ KiB [built] [code generated]

  Compiled successfully!
  Δt ____ms

[__:__:__] Finished 'js_myBundle' after ____ ms
[__:__:__] Finished 'js' after ____ ms
[__:__:__] Finished 'default' after ____ ms
",
}
`;

exports[`Compiles CSS within webpack, extracts CSS ('extractCSS' string option) 2`] = `
"!function(e,t){if("object"==typeof exports&&"object"==typeof module)module.exports=t();else if("function"==typeof define&&define.amd)define([],t);else{var o=t();for(var n in o)("object"==typeof exports?exports:e)[n]=o[n]}}(self,function(){return function(){"use strict";var e={};e.d=function(t,o){for(var n in o)e.o(o,n)&&!e.o(t,n)&&Object.defineProperty(t,n,{enumerable:!0,get:o[n]})},e.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},e.r=function(e){"undefined"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(e,"__esModule",{value:!0})};var t={};e.r(t),e.d(t,{default:function(){return o}});class o{}return t}()});
//# sourceMappingURL=myBundle.min.js.map"
`;

exports[`Compiles CSS within webpack, extracts CSS ('extractCSS' string option) 3`] = `
"!function(e,t){if("object"==typeof exports&&"object"==typeof module)module.exports=t();else if("function"==typeof define&&define.amd)define([],t);else{var o=t();for(var n in o)("object"==typeof exports?exports:e)[n]=o[n]}}(self,function(){return function(){"use strict";var e={};e.d=function(t,o){for(var n in o)e.o(o,n)&&!e.o(t,n)&&Object.defineProperty(t,n,{enumerable:!0,get:o[n]})},e.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},e.r=function(e){"undefined"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(e,"__esModule",{value:!0})};var t={};e.r(t),e.d(t,{default:function(){return o}});class o{}return t}()});
//# sourceMappingURL=myBundle.min.js.map"
`;

exports[`Compiles CSS within webpack, extracts CSS ('extractCSS' string option) 4`] = `
"a{color:#00f}.Parent{background:#fff}.Parent--before{color:#333}.Parent .Child{background:#000}.Parent--after{color:#eee}:root{--color:red}.MenuLink{color:var(--color)}
/*# sourceMappingURL=myBundle-string.min.css.map*/"
`;

exports[`Fails gracefully on broken markup 1`] = `
{
  "status": 1,
  "stdall": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            css: __PATH__/packages/integration-fixtures-cjs/fixtures/crafty-preset-lightningcss-webpack/fails/dist/css
           js: __PATH__/packages/integration-fixtures-cjs/fixtures/crafty-preset-lightningcss-webpack/fails/dist/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myBundle' ...
assets by status ___ KiB [cached] 1 asset
Entrypoint default = myBundle.min.js 1 auxiliary asset
chunk (runtime: default) myBundle.min.js (default) ___ KiB (javascript) ___ KiB (runtime) [entry] [rendered]
  dependent modules ___ KiB [dependent] 7 modules
  runtime modules ___ KiB 5 modules
  ./js/app.js + 1 modules ___ KiB [built] [code generated]

  Failed to compile.

SyntaxError: Unexpected end of input
    ...stacktrace...

  Δt ____ms ✖ 1 problem (1 error, 0 warnings)

[__:__:__] 'js_myBundle' errored after ____ ms
[__:__:__] Webpack compilation failed
[__:__:__] 'js' errored after ____ ms
[__:__:__] 'default' errored after ____ ms
",
}
`;
